src/Data/Singletons.hs

```diff
@@ -39,6 +43,9 @@ module Data.Singletons (
   KindOf, SameKind,
   SingInstance(..), SomeSing(..),
   singInstance, withSingI, withSomeSing, singByProxy, demote,
+#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 710
+  pattern SingI,
```

Comment:
Given that (logically) this pattern synonym is an extension of `singInstance`, I would put this next to `singInstance` in the export list.